tests/processor/test_loose_artifact_analyzer.py

```diff
@@ -20,79 +20,171 @@
+            os.makedirs(path)
+            expected_paths.add(path)
+            expected_archives.add(name)
+
+        create_test_archive('rpm', 'name-1-2-3.noarch.rpm')
+        create_test_archive('rpm', 'name2-1-2-3.noarch.rpm')
+        create_test_archive('container_layer', 'docker-image:sha1234')
+        create_test_archive('container_layer', 'docker-image:sha4321')
+        create_test_archive('maven', 'really-important.jar')
+        create_test_archive('maven', 'really-important-2.jar')
+
+        analyzer = LooseArtifactAnalyzer(temp_dir)
+        found_names = set()
+        found_paths = set()
+        for name, path in analyzer.unpacked_archives():
+            found_names.add(name)
+            found_paths.add(path)
+
+        assert found_names == expected_archives
+        assert found_paths == expected_paths
+
+
+def test_files_to_examine():
+    """Test the files_to_examine method correctly discoveres all files it's supposed to."""
+    expected_files = set()
+    with tempfile.TemporaryDirectory() as temp_dir:
+        def create_test_archive(expected, *args):
+            new_dir = os.path.join(temp_dir, *args[:-1])
+            try:
+                os.makedirs(new_dir)
+            except FileExistsError:
+                pass
+            f = create_test_file(new_dir, args[-1], 'content')
+            if expected:
+                expected_files.add(f)
+
+        create_test_archive(True, 'path', 'to', 'some', 'nested', 'thing.rpm')
+        create_test_archive(True, 'path', 'to', 'some', 'nested', 'thing.jar')
+        create_test_archive(False, 'path', 'to', 'some', 'nested', 'thing.txt')
+        create_test_archive(True, 'another', 'nested', 'path', 'thing.tar')
+        create_test_archive(True, 'another', 'nested', 'path', 'thing.zip')
+        create_test_archive(True, 'another', 'nested', 'path', 'thing.tar.gz')
+        create_test_archive(False, 'another', 'nested', 'path', 'thing.csv')
+
+        analyzer = LooseArtifactAnalyzer(temp_dir)
+        found_files = set()
+        for f in analyzer.files_to_examine(temp_dir):
+            found_files.add(f)
+
+        assert found_files == expected_files
 
-def test_batches():
-    """Test that the batching function works properly."""
-    analyzer = LooseArtifactAnalyzer()
-    analyzer.KOJI_BATCH_SIZE = 2
-    it = set([1, 2, 3, 4, 5])
-    ret = analyzer.batches(it)
-    assert len(ret) == 3
-    it2 = set()
-    for x, y in ret:
-        it2.add(x)
-        if y:
-            it2.add(y)
 
-    assert it == it2
+def test_local_lookup():
+    """Test the local_lookup function correctly finds Artifacts that already exist."""
+    CONTENT = 'my content'
+    SHA256_SUM = '47a96905708e5470528752169f80e1d8d8b79c599ed35b0979fb9f17e9babfe6'
 
+    checksum_node = content.Checksum.get_or_create({
+        'algorithm': 'sha256',
+        'checksum': SHA256_SUM,
+        'checksum_source': 'unsigned'})[0]
```

Comment:
Can also use `ChecksumFactory` here. In fact, even this spaghetti code works:
```
ArtifactFactory.create().checksums.connect(ChecksumFactory.create(checksum=SHA256_SUM))
```

Reply:
Fair enough, although I'll use multiple lines for it.